### DISCUSSION TITLE
Swiper JS autoplay when in viewport

### DISCUSSION THREAD

**Discussion by pcgreen-john** [nolimits4web/swiper#6033] [Q&A] [answered]
Hello and sorry if this has been answered already (couldn't find anything on search or google) or if it's extremely simple but I can't seem to make it work.
My JS knowledge is limited but I can navigate around code and do changes and "connect the dots".
Swiper JS has been amazing to fit my clients' needs in cooperation with Oxygen builder but I have a seemingly simple request which I haven't managed to find an answer for the past days:

**How do I make Swiper JS autoplay start, when it appears in viewport?**

There are elements which are located in the middle of the page and by the time someone scrolls there, it's already on 2nd or 3rd slide.
I've tried _jquery appear_, _lazy loading_, nothing works for me.

My current script is this (without lazy loading):

	var swiper = new Swiper('.swiper-container', {
		loop: true,
		pagination: {
			el: '.swiper-pagination',
			clickable: true,
		},
		   	navigation: {
		    	nextEl: '.swiper-button-next',
		    	prevEl: '.swiper-button-prev',
		  	  },
		autoplay: {
			delay: 4000,
			disableOnInteraction: true,
		},
		noSwiping: false,
		slidesPerView: 1.2,
		grabCursor:true,
		effect: slide
	});

Any help is really appreciated, as I'm using swiper on multiple projects.
Preferably with an example since I'm a JS beginner!
Thanks!

**Comment by hrishikesh-k** [comment, marked as answer]
You can use InersectionObserver to observe and initialize Swiper only when someone scrolls to that element:

```js
new IntersectionObserver((entries, observer) => {
  if (entries[0].isIntersecting) {
    observer.disconnect()
    // your new Swiper part
  }
}).observe(document.querySelector('.swiper-container'))
```

This might not be the best solution as it won't even initialise Swiper for the element till someone scrolls to it, so I'm not sure if it fits your needs, but it works in my use case - might give you some ideas.

**Reply by pcgreen-john** [reply]
Thank you, thank you, THANK YOU!
This was exactly what I was looking for. I just used as target the div wrapper for the slider and some other elements, and used Intersection Observer as you suggested to enable autoplay once I scrolled to this element, like so

	new IntersectionObserver((entries, observer) => {
		if (entries[0].isIntersecting) {
			observer.disconnect();
			swiper.autoplay.start();
		}
	}).observe(document.querySelector('#slider-projects'));	

I'm using the old autoplay function since Oxygen runs on 5.X Swiper and the latest one for some reason expanded the slider to 100% width while I only wanted it 80% aligned to right.

Thanks again!
